packages/shared/src/components/modals/NewRankModal.tsx

```diff
@@ -73,24 +75,40 @@ export default function NewRankModal({
 
   useEffect(() => {
```

Comment:
I feel like we have this kinda low-level logic of handling listener and timeouts. My first idea was to have useAnimateRank hook with this logic hidden, but I'm not sure it can be done in a clear way (one of the reasons is that we have state updates inside), so it's just more like a side note

Reply:
Yes, but I think there is nothing wrong with hooks used only in one place if they help to keep components more readable and hide implementation details from the component itself, so hook works like a contract for a component and can be implemented in different ways.

Disclaimer - example is abstract and I'm not talking about what we have in this PR and I just want to explain how I feel about hooks. 

Example:

```
// @hooks/useAnimateRank


// different possible ways of implementation
// 1. Using refs and timeouts 
// 2. Using some animation library
export const useAnimateRank = () => {
// Exact implementation here
}

```

```
// RankComponent.tsx

import useAnimateRank from '@hooks/useAnimateRank';

...
...

const RankComponent = () => {
   
   // Implementation can be different 
   // But hook is used only in this way
   // So no changes in component required when hook implementation changed
   useAnimateRank();

   return <>{/*component JSX here*/}</>
};

...
...

```

Hope it makes sense, but for now we are good with what you just implemented

Reply:
Totally and 100% agree this would be amazing to rework, let me add a new backlog item for it.